lib/Frontend/Frontend.cpp

```diff
@@ -261,175 +264,281 @@ ModuleDecl *CompilerInstance::getMainModule() {
   return MainModule;
 }
 
-void CompilerInstance::performSema() {
-  const FrontendOptions &options = Invocation.getFrontendOptions();
-  const InputFileKind Kind = Invocation.getInputKind();
-  ModuleDecl *MainModule = getMainModule();
-  Context->LoadedModules[MainModule->getName()] = MainModule;
-
-  auto modImpKind = SourceFile::ImplicitModuleImportKind::Stdlib;
+static void
+addAdditionalInitialImportsTo(SourceFile *SF, CompilerInstance::ImplicitImports &implicitImports) {
+  if (!implicitImports.objCModuleUnderlyingMixedFramework && !implicitImports.headerModule &&
+      implicitImports.modules.empty())
+    return;
```

Comment:
I don't think we get a win from this early exit. It's just more code to maintain if we add another case.

Reply:
Is there some way to get Tower to git-clang-format automatically?

Reply:
The early exit was there before I messed with the code, but I agree with you about it. Glad you caught it. Increases my feeling that this refactoring is worth it.